service/spawn.go

```diff
@@ -469,3 +467,107 @@ func (uis *UIServer) modifySpawnHost(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 }
+
+func (uis *UIServer) modifyVolume(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	updateParams := restModel.APIVolumeModify{}
+	if err := utility.ReadJSON(util.NewRequestReader(r), &updateParams); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+	}
+
+	volumeID := gimlet.GetVars(r)["volume_id"]
+	vol, err := host.FindVolumeByID(volumeID)
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrapf(err, "error finding volume '%s'", volumeID))
+		return
+	}
+	if vol == nil {
+		uis.LoggedError(w, r, http.StatusNotFound, errors.Wrapf(err, "no volume '%s' exists", volumeID))
+		return
+	}
+
+	u := MustHaveUser(r)
+	if u.Username() != vol.CreatedBy {
+		uis.LoggedError(w, r, http.StatusUnauthorized, errors.New("not authorized to modify this volume"))
+		return
+	}
+
+	if updateParams.Action == nil {
+		uis.LoggedError(w, r, http.StatusBadRequest, errors.New("no action specified"))
+		return
+	}
+
+	mgrOpts := cloud.ManagerOpts{
```

Comment:
Can you do this only if the action isn't VolumeRename?

Reply:
done